### ISSUE TITLE
Micro_speech error when given new model

### ISSUE THREAD

**Issue by michealcarac** [tensorflow/tensorflow#44868]
**System information**
- Have I written custom code : Only changed the model and everything the model requires (size of it)
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Nano 33 BLE Sense 
- TensorFlow installed from (source or binary): Arduino Library
- TensorFlow Version: 2.1.0-ALPHA

When ran I get this error:
```
Feature generation failed
Requested feature_data_ size 536907080 doesn't match 1960

```
It should be the basic micro_speech output with words found and time of them found.
It was originally working with this model, but for some reason, it is not longer working. 

Code: https://github.com/michealcarac/TempMicroSpeech

In micro_speechwithfinal.ino:
```
int8_t feature_buffer[kFeatureElementCount];

 kFeatureElementCount = (kFeatureSliceSize * kFeatureSliceCount);

```
in micro_features_micro_model_settings.h

However, 
when 
```
static FeatureProvider static_feature_provider(kFeatureElementCount,
                                                 feature_buffer);
```
in  micro_speechwithfinal.ino

then in feature_provider.cpp
```

FeatureProvider::FeatureProvider(int feature_size, int8_t* feature_data)
    : feature_size_(feature_size),
      feature_data_(feature_data),
      is_first_run_(true) {
  // Initialize the feature data to default values.
  for (int n = 0; n < feature_size_; ++n) {
    feature_data_[n] = 0;
  }
}

```

BUT for some reason, feature_size_ is 536907080 yet kFeatureElementCount was 1960 in the beginning...

**Comment by ymodak** [comment]
Please take a look at this thread https://github.com/tensorflow/tensorflow/issues/39938

**Comment by michealcarac** [comment]
Issue fixed by creating a new model on TF 1.15.

**Comment by ghanko** [comment]
It seems that model created with newer tf uses Conv2D op too and the example is not up-to-date with this. Adding
`if (micro_op_resolver.AddConv2D() != kTfLiteOk) {
    return;
  }`
  resolved the issue for me.
  Don't forget to increase opCount of MicroMutableOpResolver if a new op is added:
  `static tflite::MicroMutableOpResolver<5> micro_op_resolver(error_reporter);`

**Comment by Ashment** [comment]
Hello @ghanko 

I have the same problem as in this thread, where I get this error:

```
Feature generation failed
Requested feature_data_ size 536907080 doesn't match 1960
```

I've tried your solution by adding the conv2d op to the micro op resolver, and it did not fix the issue. I've also tried increasing the arena size, as well as swapping to an all ops resolver, neither of which fixed the issue either. Maybe you have more insight into how to fix this? Thanks!

**Comment by preston-Glenn** [comment]
Just encountered the same issue and tried both suggested fixes? Are there any other plausible issues to consider?

**Comment by rzarhmi** [comment]
> It seems that model created with newer tf uses Conv2D op too and the example is not up-to-date with this. Adding `if (micro_op_resolver.AddConv2D() != kTfLiteOk) { return; }` resolved the issue for me. Don't forget to increase opCount of MicroMutableOpResolver if a new op is added: `static tflite::MicroMutableOpResolver<5> micro_op_resolver(error_reporter);`

Thank you so much! This fixed my issue!

**Comment by avisinghal6** [comment]
> It seems that model created with newer tf uses Conv2D op too and the example is not up-to-date with this. Adding `if (micro_op_resolver.AddConv2D() != kTfLiteOk) { return; }` resolved the issue for me. Don't forget to increase opCount of MicroMutableOpResolver if a new op is added: `static tflite::MicroMutableOpResolver<5> micro_op_resolver(error_reporter);`

Thank you so much, This worked for me as well.

**Comment by Lieber0402** [comment]
> It seems that model created with newer tf uses Conv2D op too and the example is not up-to-date with this. Adding `if (micro_op_resolver.AddConv2D() != kTfLiteOk) { return; }` resolved the issue for me. Don't forget to increase opCount of MicroMutableOpResolver if a new op is added: `static tflite::MicroMutableOpResolver<5> micro_op_resolver(error_reporter);`

This works for me too, thanks a lot. But I'm wondering why the deployment on the develop board Sparkfun Edge didn't work well. It can't recognize the keywords "yes" and "no".

**Comment by rishita-asl** [comment]
Facing the same issue here, the suggested change `if (micro_op_resolver.AddConv2D() != kTfLiteOk) { return; }` didnt work for me. I cant event connect to the borad using serial monitor after adding the Conv2D. I had increased the opCount as well Arena size but still facing the same issue. I tried building the model with 4 keywords instead of 2 like the example. Any help would be appreciated.

**Comment by tjpeter** [comment]
I am still having the same issue. I have added the additional ops, e.g. 
`if (micro_op_resolver.AddConv2D() != kTfLiteOk) { return; }`

and also increased the opCount of MicroMutableOpResolver, e.g.
`static tflite::MicroMutableOpResolver<5> micro_op_resolver(error_reporter);`

Yet, I still get 
`Requested feature_data_size 536907080 does not match 1024
`
I would appreciate any help-

**Comment by GonzalezMada** [comment]
I think the problem is with the operators used in the model.
The operators used by the TensoFlow team to create the microspeech project example are: DepthwiseConv2D, Reshape, FullyConnected, and Softmax. We can see that on the code below:
```
static tflite::MicroMutableOpResolver<4> micro_op_resolver;
  if (micro_op_resolver.AddDepthwiseConv2D() != kTfLiteOk) {`
    return;
 }
  if (micro_op_resolver.AddFullyConnected() != kTfLiteOk) {
    return;
  }
  if (micro_op_resolver.AddSoftmax() != kTfLiteOk) {
    return;
  }
  if (micro_op_resolver.AddReshape() != kTfLiteOk) {
    return;
  }
```

I have trained a model using the [colab](https://github.com/tensorflow/tflite-micro/blob/main/tensorflow/lite/micro/examples/micro_speech/train/train_micro_speech_model.ipynb) provided by @pjpratik. Then deploy the model on Arduino Nano 33 BLE sense REV2. And I got the same issue.
```
Feature generation failed
Requested feature_data_ size 536907080 doesn't match 1960
```

**SOLUTION:**

I have visualized the tflite model using [Neutron](https://netron.app/), and I saw that the NEW model uses these operators: Conv2D, Reshape, FullyConnected, and Softmax. So, the operators in our code for deployment should include these new operators.

 I have changed the operator declaration to:
```
static tflite::MicroMutableOpResolver<4> micro_op_resolver;
  if (micro_op_resolver.AddFullyConnected() != kTfLiteOk) {
    return;
  }
  if (micro_op_resolver.AddSoftmax() != kTfLiteOk) {
    return;
  }
  if (micro_op_resolver.AddReshape() != kTfLiteOk) {
    return;
  }
  if (micro_op_resolver.AddConv2D() != kTfLiteOk) { 
    return; } // I have added the Conv2D op and deleted the DepthwiseConv2D op.
```

**AND IT WORKS**
